systemtest/pom.xml

```diff
@@ -291,8 +297,35 @@
                             </ignoredUnusedDeclaredDependencies>
                         </configuration>
                     </execution>
+                    <execution>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>copy-dependencies</goal>
+                        </goals>
+                        <configuration>
+                            <outputDirectory>${project.build.directory}/lib</outputDirectory>
+                        </configuration>
+                    </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>io.skodjob</groupId>
+                <artifactId>test-docs-generator-maven-plugin</artifactId>
+                <version>${skodjob-doc.version}</version>
+                <executions>
+                    <execution>
+                        <phase>post-integration-test</phase>
+                        <goals>
+                            <goal>test-docs-generator</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <testsPath>${project.basedir}/src/test/java/io/strimzi/systemtest</testsPath>
+                    <docsPath>${project.basedir}/docs/</docsPath>
```

Comment:
Why `/docs/`? Yuo should probably explain a bit more how this will work.

Reply:
This is basically the main consideration I have: whether we should go for:

1. Generating documentation in `/systemtest/docs`, where one could then review how our system tests function from the documentation. This documentation would always be synced with the code docs, i.e., @SuiteDoc and @TestDoc. The primary purpose is to provide a source of truth between developers and testers during test triage.
2. The second approach would be to consider other options for where we could generate this documentation.
3. The third option is to let end-users generate such documentation.

and for me it was the first approach, which gave me the most sense...but that's not only up to me so maybe:
@strimzi/system-test-contributors have better idea ;)

Reply:
Would `developer-docs/...` be a better place to have it? There are already docs for how to test etc. In the system tests it would be completely lost. If you plan to have it committed (which for me is fine), you should have an azure pipelines check for the files to be up to date similarly to what we have for `packaging/`.

Reply:
As I mentioned offline, I'm not sure if it should be part of the operators repository, as it can be a lot of lines and files that you would pull -> and I mentioned that maybe it can be stored in different repository and we could point to that repository in our README. But if everyone will be fine with having it inside the operators repository, then fine -> it should be in the `developer-docs/...` as Jakub mentioned

Reply:
Separate repo will bring it out of sync, would not allow to track releases, etc. I think the option is to store it in this repo, or not store it and have people generate it when needed from the code.

Reply:
@Frawless @henryZrncik @jankalinic any opinion what could be better (having it in the repo or leaving it for users to generate it)?

Reply:
I guess that once we will have all docs merged new changes will not requiere that much changes; but if it seems like a lot letting user generate it seems to me as perfectly good option.

Reply:
Okay, so I guess 1. approach (i.e., generating such doc here) and specifically in `developers-docs/` would be the best fit. I will also implement that Azure check which you mentioned Jakub ;)

Reply:
+1 for store it in repo and +1 for checking if the files are in sync. It would be great to have check that the docs are generated for new tests, but that's something we cannot do until all tests will be documented and even after I am not sure how it will work. Guess we can revisit it later.

Reply:
I have updated this, please check it :)